NEXT_CHANGELOG.md

```diff
@@ -26,9 +26,27 @@ By [@USERNAME](https://github.com/USERNAME) in https://github.com/apollographql/
 
 ## 🛠 Maintenance
 
+### CI: Enable compliance checks /except/ licenses.html update ([Issue #2514](https://github.com/apollographql/router/issues/2514))
+
+In [#1573](https://github.com/apollographql/router/pull/1573), we removed the compliance checks for non-release CI pipelines, because the cargo-about output would change ever so slightly.
+Some checks however are very important and prevent us from inadvertently downgrading libraries and needing to open [#2512](https://github.com/apollographql/router/pull/2512).
+
+This PR does the following changes:
+- Introduce `cargo xtask licenses` to check and update licenses.html.
+- Separate compliance (cargo-deny, which includes license checks) and licenses generation (cargo-about) in xtask
+- Enable compliance as part of our CI checks for each open PR
+- Update `cargo xtask all` so it runs tests, checks compliance and licenses
+- Introduce `cargo xtask dev` so it checks compliance and runs tests
+
+Use `cargo xtask all` before a release.
+Use `cargo xtask dev` before a PR.
+
+Updating licenses.html is now driven by `cargo xtask licenses`, which is part of the release checklist.
+
+By [@o0Ignition0o](https://github.com/o0Ignition0o) in https://github.com/apollographql/router/pull/XXX_TBD
+
```

Comment:
I like how you wrote a great PR message and were able to re-use it here.  I would just replace the word "PR"  with "change" in the changelog since it will not be read in the context of a PR. (nit though.)

Also, once you merge this in with `dev`, as @BrynCooke already noted, you'll need to use the new pattern from https://github.com/apollographql/router/pull/2534.